internal/resolver/dns/dns_resolver.go

```diff
@@ -86,24 +86,24 @@ var (
 	minDNSResRate = 30 * time.Second
 )
 
-var customAuthorityDialler = func(authority string) func(ctx context.Context, network, address string) (net.Conn, error) {
+var customHostDialler = func(urlHost string) func(ctx context.Context, network, address string) (net.Conn, error) {
```

Comment:
Nit: `Dialer`.

I'm not sure about this renaming though...  We used to call it "authority", which was the full `[user@]host[:port]` of the target (see https://www.rfc-editor.org/rfc/rfc3986#section-3.2).  It actually used to include all of that when we manually parsed, but nobody should have ever used `user@`.  Now we use the Go `net/url` package which turns the target into a [url.URL](https://pkg.go.dev/net/url#URL).  Technically "`authority`" would include `url.User`.  But this is still stuff from the authority field.

Maybe `newNetResolver` for `customHostResolver` and `addressDialer` instead or `customHostDialler`?